src/config/wmodules-gcp.c

```diff
@@ -0,0 +1,221 @@
+                return OS_INVALID;
+            }
+            os_strdup(nodes[i]->content, gcp->subscription_name);
+        }
+        else if (!strcmp(nodes[i]->element, XML_CREDENTIALS_FILE)) {
+            if(strlen(nodes[i]->content) >= PATH_MAX) {
+                merror("File path is too long. Max path length is %d.", PATH_MAX);
+                return OS_INVALID;
+            } else if (strlen(nodes[i]->content) == 0) {
+                merror("Empty content for tag '%s' at module '%s'.", XML_CREDENTIALS_FILE, WM_GCP_CONTEXT.name);
+                return OS_INVALID;
+            }
+
+            char realpath_buffer[PATH_MAX] = {0};
+
+            if(nodes[i]->content[0] == '/') {
+                sprintf(realpath_buffer, "%s", nodes[i]->content);
+            } else {
+                char relative_path[PATH_MAX] = {0};
+
+                sprintf(relative_path, "%s/", DEFAULTDIR);
+                strcat(relative_path, nodes[i]->content);
+                const char * const realpath_buffer_ref = realpath(relative_path, realpath_buffer);
+                if (!realpath_buffer_ref) {
+                    mwarn("File '%s' from tag '%s' not found.", realpath_buffer, XML_CREDENTIALS_FILE);
+                    return OS_INVALID;
+                }
+            }
+
+            // Beware of IsFile inverted, twisted logic.
+            if (IsFile(realpath_buffer)) {
+                mwarn("File '%s' not found. Check your configuration.", realpath_buffer);
+                return OS_INVALID;
+            }
+
+            os_strdup(realpath_buffer, gcp->credentials_file);
+        }
+        else if (!strcmp(nodes[i]->element, XML_INTERVAL)) {
+            char *endptr;
+            gcp->interval = strtoul(nodes[i]->content, &endptr, 0);
+
+            if (gcp->interval == 0 || gcp->interval == UINT_MAX) {
+                merror("Invalid interval value.");
+                return OS_INVALID;
+            }
+
+            switch (*endptr) {
+                case 'w':
+                    gcp->interval *= 604800;
+                    break;
+                case 'd':
+                    gcp->interval *= 86400;
+                    break;
+                case 'h':
+                    gcp->interval *= 3600;
+                    break;
+                case 'm':
+                    gcp->interval *= 60;
+                    break;
+                case 's':
+                case '\0':
+                    break;
+                default:
+                    merror("Invalid interval value.");
+                    return OS_INVALID;
+            }
+        }
+        else if (!strcmp(nodes[i]->element, XML_MAX_MESSAGES)) {
+            if (!nodes[i]->content) {
+                merror("Empty content for tag '%s'", XML_MAX_MESSAGES);
+                return OS_INVALID;
+            }
+
+            char *endptr;
+            gcp->max_messages = strtoul(nodes[i]->content, &endptr, 0);
```

Comment:
Having max_messages configured to something other than a number does not issue a warning/error. The program still works because a configuration of 0 is assumed and the wodle is run with the default value of 100, but it would be nice to have some feedback on a wrong configuration value.